choccy_chip/src/emulator/opcode.rs

```diff
@@ -394,9 +394,25 @@ impl Emu {
         }
     }
 
-    // TODO: Implement this
+    /// Handle a keyop wait operation
+    /// Waits for a key press and stores the key in the given register.
+    /// #Arguments
+    /// - `reg_id`: The register to store the key in.
+    /// #Notes
+    /// - This is a blocking operation.
     fn handle_keyop_wait(&mut self, reg_id: u8) {
-        todo!()
+        let mut pressed = false;
```

Comment:
can you add that we just decided to pick the lowest key that comes first for blocking.

Reply:
lgtm